Update .NET Framework to 4.8, migrate to sdk format

Repiteo opened the pull request:
With .NET Core being in the air, this is something to bridge the gap in the meantime to allow modern Visual Studio instances to properly test and debug sharpshell as-is. Preliminary testing makes it seem like .NET Standard 2.0 could be another stopgap solution for the `SharpShell` logic specifically (the GUI stuff is another story), but that wouldn't be as smooth of a transition without first making sure everything functions as-is.

Repiteo committed:
Update .NET Framework versions (4.0/4.5 → 4.8)

Repiteo committed:
Convert `.csproj` files to sdk format

• Make `Any CPU` the only build target, as `SharpShellNativeBridge` is no longer part of the main Solution

osexpert commented on SharpShell/Tools/ServerRegistrationManager/ServerRegistrationManager.csproj:
```diff
@@ -1,110 +1,37 @@
+    <PackageReference Include="System.Data.DataSetExtensions" Version="4.5.0" />
+  </ItemGroup>
+  <Target Name="PreBuild" BeforeTargets="PreBuildEvent">
+    <Exec Command="xcopy $(SolutionDir)artifacts\build\SharpShell\SharpShell.dll $(ProjectDir)EmbeddedReferences\ /F /R /Y /I " />
```
missing quotes here

osexpert replied:
```suggestion
    <Exec Command="xcopy &quot;$(SolutionDir)artifacts\build\SharpShell\SharpShell.dll&quot; &quot;$(ProjectDir)EmbeddedReferences\&quot; /F /R /Y /I" />
```

osexpert commented:
Visual Studio 2017 is most likely too old. Should probably change appveyor.yml from
image: Visual Studio 2017
to
image: Visual Studio 2022

Repiteo committed:
Modernize builder tools

osexpert commented:
I think you need to switch to use dotnet build instead of msbuild in build.ps1 (or maybe not required, but IIRC msbuild and sdk style csproj are not best friends and will fight from time to time)

osexpert commented on appveyor.yml:
```diff
@@ -1,49 +1,57 @@
-    nuget restore SharpShell\SharpShell.sln
-    .\SharpShell\config.ps1
+  - ps: |
+      nuget restore SharpShell/SharpShell.sln
```
I think dotnet build will do package restore automagically...so if/when switching to dotnet build, this (nuget restore) may not be required anymore

Repiteo replied:
The endgoal is switching to dotnet, so this makes sense to integrate. Didn't think that it would work if on the .NET Framework, but seeming as it calls to MSBuild anyway it *should* be fine.

Repiteo replied:
Hmm, doesn't seem like it'll work. Not 100% sure why that is, but it seems `dotnet` isn't a drop-in replacement for `msbuild` in this context.

Repiteo replied:
Alright, I've figured it out! A few other settings had to be adjusted to be suitable for a dotnet environment, so applying that lets SharpShell build without an issue. However, as SharpShellNativeBridge will still require MSBuild no matter what, I've separated the jobs into dedicated dotnet & msbuild sections. Now the overall time spent building is much shorter!

Repiteo committed:
Implement `Directory.Build.props` for shared info

• Remove all AssemblyInfo, migrate information to project when relevant
• Remove legacy files that no longer link via new systems

Repiteo committed:
Modernize NuGet packages, update dotnet tests

• Fix build warnings

osexpert commented on SharpShell/Dependencies/Apex/Apex.dll:
Why is this file smaller? Or is it removed completely?

osexpert replied:
Seems related to 
https://github.com/dwmkerr/apex
https://www.nuget.org/packages/Apex
But file version is 1.6.0, so may be a private build that was never published to nuget, because last nuget is 1.5.0...

osexpert replied:
But is it possible that Apex.WinForms.dll loads Apex.dll dynamically? Or is it truly unused?

Repiteo replied:
Everything certainly *seems* to function, but the file version is a good point. No clue why Apex.WinForms was never put on nuget
